File: src/services/sticker.go
package services

import (
	"encoding/json"
	console "tomoribot-geminiai-version/src/thirdpartyLangs"
)

type StickerResult struct {
	Success   bool  `json:"success"`
	ImagePath string `json:"imagePath"`
}

func AddExifOnSticker(exifPath string, packname string, author string) (StickerResult, error) {
	args := []string{
		"./nodejs/sticker.js",
		"--addexif",
		exifPath,
		packname,
		author,
	}
	result, err := console.RunNodeCommand(args)
	if err != nil {
		return StickerResult{}, err
	}
	var stickerResult interface{}
	err = json.Unmarshal(result.Bytes(), &stickerResult)
	if err != nil {
		return StickerResult{}, err
	}
	var stickerResults StickerResult
	stickerResults.Success = stickerResult.(map[string]interface{})["success"].(bool)
	if !stickerResults.Success {
		return stickerResults, nil
	}
	stickerResults.ImagePath = stickerResult.(map[string]interface{})["imagePath"].(string)
	return stickerResults, nil
}

File: nodejs/sticker.js
const yargs = require("yargs");
const { Sticker } = require("./images/images");
const argv = yargs
    .option("addexif", {
        alias: "exif",
        description: "Add exif on webp sticker",
        type: "string",
        array: true
    })
    .help()
    .alias("help", "h")
    .argv;

if (argv.addexif) {
    const sticker = new Sticker(argv.addexif[0])
    sticker.build({
        packname: argv.addexif[1],
        author: argv.addexif[2]
    }).then(result => {
        console.log(JSON.stringify({
            success: true,
            imagePath: result
        }))
    }).catch(JSON.stringify({
      success: false
  }))
}